src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -586,9 +587,17 @@ def pre_remove(self, daemon: DaemonDescription) -> None:
         super().pre_remove(daemon)
 
         assert daemon.daemon_id is not None
+        assert daemon.hostname is not None
+        assert self.mgr._cluster_fsid is not None
         daemon_id: str = daemon.daemon_id
+        service_name: str = daemon.service_name
+        fsid: str = self.mgr._cluster_fsid
         self._check_safe_to_destroy(daemon_id)
 
+        logger.info('shutting down %s' % service_name)
+        # stop the deamon before we remove and destroy the daemon
+        cmd = "sudo systemctl stop ceph-%s@%s" % (fsid, service_name)
+        os.system(cmd)
```

Comment:
@adk3798 I don't think this will actually work, currently, I don't think there is a MON or other ceph cli command that allows us to stop the monitor, there is only `service ceph -a stop mon.{mon-id}` or using `systemctl`.  These are all executed through cmd and not ceph cli, do you mind guiding me to how I can execute cmd commands inside this part of cephadm? Thanks!

Reply:
I'd think a change like this to the way we shutdown the daemon belong smore in https://github.com/ceph/ceph/blob/master/src/cephadm/cephadm#L6446 than here in the mgr module. All daemon removals run through that code path. Although it looks like it already run systemctl stop. Either way, that's where you'd want to look.

Reply:
thanks! will take a look